lib/ansible/modules/storage/ibm/ibm_svc_host.py

```diff
@@ -0,0 +1,347 @@
+    type:
+        description:
+            - This is host type
+        default:
+        type: str
+    log_path:
+        description:
+            - For extra logging
+        type: str
+    validate_certs:
+        description:
+            - validate_certs
+        type: bool
+author:
+    - Chun Yao (@chunyao)
+'''
+
+EXAMPLES = '''
+- name: Define a new host
+  ibm_svc_host:
+        clustername: mcr-tb5-cluster-03
+        domain: stglab.manchester.uk.ibm.com
+        username: superuser
+        password: passw0rd
+        log_path: /tmp/playbook.debug
+        name: host4test
+        state: present
+        fcwwpn: 100000109B570216
+        iscsiname: iqn.1994-05.com.redhat:2e358e438b8a
+        iogrp: 0:1:2:3
+        protocol: scsi
+        type: generic
+
+- name: Delete host
+  ibm_svc_host:
+        clustername: mcr-tb5-cluster-03
+        domain: stglab.manchester.uk.ibm.com
+        username: superuser
+        password: passw0rd
+        log_path: /tmp/playbook.debug
+        name: host4test
+        state: absent
+'''
+
+RETURN = '''
+'''
+
+import logging
+from traceback import format_exc
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ibm_svc_utils import IBMSVCRestApi, svc_argument_spec
+from ansible.module_utils._text import to_native
+
+
+class IBMSVChost(object):
+    def __init__(self):
+        argument_spec = svc_argument_spec()
+
+        argument_spec.update(
+            dict(
+                name=dict(type='str', required=True),
+                state=dict(type='str', required=True, choices=['absent',
+                                                               'present']),
+                fcwwpn=dict(type='str', required=False),
+                iscsiname=dict(type='str', required=False),
+                iogrp=dict(type='str', default='0:1:2:3'),
+                protocol=dict(type='str', default='scsi', choices=['scsi',
+                                                                   'nvme',
+                                                                   'iscsi']),
+                type=dict(type='str')
+            )
+        )
+
+        self.module = AnsibleModule(argument_spec=argument_spec,
+                                    supports_check_mode=True)
+
+        # logging setup
+        log_path = self.module.params['log_path']
+        self._logger = logging.getLogger(self.__class__.__name__)
+        self.debug = self._logger.debug
+        if log_path:
+            logging.basicConfig(level=logging.DEBUG, filename=log_path)
+
+        # Required
+        self.name = self.module.params['name']
+        self.state = self.module.params['state']
+
+        # Optional
+        self.fcwwpn = self.module.params['fcwwpn']
+        self.iscsiname = self.module.params['iscsiname']
+        self.iogrp = self.module.params['iogrp']
+        self.protocol = self.module.params['protocol']
+        self.type = self.module.params['type']
```

Comment:
If those are optional, wouldn't you rather use some safe way to check their value before passing? Or a default value? not all of them have a default value

Reply:
@tzurE Thank you very much, fixed it, please help to review it.